securedrop/journalist_app/api.py

```diff
@@ -153,7 +153,7 @@ def single_source(source_uuid: str) -> Tuple[flask.Response, int]:
             utils.delete_collection(source.filesystem_id)
             return jsonify({'message': 'Source and submissions deleted'}), 200
         else:
-            abort(404)
+            abort(405)
```

Comment:
Nice, though as long as we've correctly defined the allowed methods in `@api.route`, Flask will do this. You've made me realize, though, that it does not set the `Allow` header as required by the spec. Other weaknesses in our error handling have come up lately, so I think I'm going to file another issue and suggest we address this there.

Reply:
This change was based on your comment at https://github.com/freedomofpress/securedrop/pull/5464#discussion_r492294392

Reply:
:facepalm: Right. Forgot about `HEAD` being implicitly allowed. I'll try to remember to gather all these into that forthcoming issue.